runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java

```diff
@@ -89,24 +89,36 @@ public void setCurrentTransform(AppliedPTransform<?, ?, ?> transform) {
     this.currentTransform = transform;
   }
 
-  public <T extends PInput> T getInput(PTransform<T, ?> transform) {
+  public <T extends PValue> T getOnlyInput(PTransform<T, ?> transform) {
     checkArgument(currentTransform != null && currentTransform.getTransform() == transform,
         "can only be called with current transform");
     @SuppressWarnings("unchecked")
-    T input = (T) currentTransform.getInput();
+    T input = (T) Iterables.getOnlyElement(currentTransform.getInputs()).getValue();
```

Comment:
Couldn't you simply call `getInput()` instead of `currentTransform.getInputs()` and it will validate so you could drop the `checkArguemnts` as well. True for `getOnlyOutput()` as well.